src/tests/test_classifier_workflow_service.py

```diff
@@ -44,6 +45,69 @@ def test_with_new_file(self, config, tmp_path):
     self.assert_row_matches(rows[4], img_paths[3], config.output_base_path / 'ncc/fcpl_005.jpg', '0', 0.6976)
     assert len(rows) == 5 # includes header row
 
+  def test_with_resume(self, config, tmp_path):
+    torch.manual_seed(42) 
+    report_path = tmp_path / 'report.csv'
+    img_paths = [Path.cwd() / 'fixtures/normalized_images/gilmer/00276_op0204_0001.jpg',
+        Path.cwd() / 'fixtures/normalized_images/ncc/G3902-F3-1981_U5_front.jpg']
+
+    subject = ClassifierWorkflowService(config, report_path)
+    subject.process(img_paths)
+
+    rows = self.load_csv_rows(report_path)
+    self.assert_row_matches(rows[1], img_paths[0], config.output_base_path / 'gilmer/00276_op0204_0001.jpg', '1', 0.8863)
+    self.assert_row_matches(rows[2], img_paths[1], config.output_base_path / 'ncc/G3902-F3-1981_U5_front.jpg', '1', 0.9409)
+    assert len(rows) == 3 # includes header row
+
+    progress_list = self.load_progress_list(tmp_path)
+    assert progress_list[0] == str(img_paths[0])
+    assert progress_list[1] == str(img_paths[1])
+    assert len(progress_list) == 2
+
+    img_paths2 = [Path.cwd() / 'fixtures/normalized_images/gilmer/00276_op0204_0001.jpg',
+        Path.cwd() / 'fixtures/normalized_images/ncc/G3902-F3-1981_U5_front.jpg',
+        Path.cwd() / 'fixtures/normalized_images/ncc/Cm912m_U58b9.jpg',
+        Path.cwd() / 'fixtures/normalized_images/ncc/fcpl_005.jpg']
+
+    subject2 = ClassifierWorkflowService(config, report_path)
+    subject2.process(img_paths2)
+
+    rows = self.load_csv_rows(report_path)
+    self.assert_row_matches(rows[1], img_paths2[0], config.output_base_path / 'gilmer/00276_op0204_0001.jpg', '1', 0.8863)
+    self.assert_row_matches(rows[2], img_paths2[1], config.output_base_path / 'ncc/G3902-F3-1981_U5_front.jpg', '1', 0.9409)
+    self.assert_row_matches(rows[3], img_paths2[2], config.output_base_path / 'ncc/Cm912m_U58b9.jpg', '1', 0.9775)
+    self.assert_row_matches(rows[4], img_paths2[3], config.output_base_path / 'ncc/fcpl_005.jpg', '0', 0.6976)
+    assert len(rows) == 5 # includes header row
+
+    progress_list = self.load_progress_list(tmp_path)
+    assert progress_list[0] == str(img_paths2[0])
+    assert progress_list[1] == str(img_paths2[1])
+    assert progress_list[2] == str(img_paths2[2])
+    assert progress_list[3] == str(img_paths2[3])
+    assert len(progress_list) == 4
+
+  def test_restart(self, config, tmp_path):
+    torch.manual_seed(42) 
+    report_path = tmp_path / 'report.csv'
+    img_paths = [Path.cwd() / 'fixtures/normalized_images/gilmer/00276_op0204_0001.jpg']
+
+    subject = ClassifierWorkflowService(config, report_path)
+    subject.process(img_paths)
+
+    subject2 = ClassifierWorkflowService(config, report_path, restart = True)
+    subject2.process(img_paths)
+
+    rows = self.load_csv_rows(report_path)
+    # Duplicate entry should be present, since we told it to restart the progress tracking
```

Comment:
This works as intended on Longleaf, but is having duplicated rows in the results .csv necessarily the desired behavior?